% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
\documentclass[
]{article}
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{amsmath,amssymb}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\newsavebox\pandoc@box
\newcommand*\pandocbounded[1]{% scales image to fit in text height/width
  \sbox\pandoc@box{#1}%
  \Gscale@div\@tempa{\textheight}{\dimexpr\ht\pandoc@box+\dp\pandoc@box\relax}%
  \Gscale@div\@tempb{\linewidth}{\wd\pandoc@box}%
  \ifdim\@tempb\p@<\@tempa\p@\let\@tempa\@tempb\fi% select the smaller of both
  \ifdim\@tempa\p@<\p@\scalebox{\@tempa}{\usebox\pandoc@box}%
  \else\usebox{\pandoc@box}%
  \fi%
}
% Set default figure placement to htbp
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\usepackage{bookmark}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={Naja\_synteny\_VenomGenes},
  pdfauthor={Alex},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{Naja\_synteny\_VenomGenes}
\author{Alex}
\date{2025-11-28}

\begin{document}
\maketitle

\section{Workflow to compare venom coding regions of naja
naja}\label{workflow-to-compare-venom-coding-regions-of-naja-naja}

We start with two prerequesites:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Annotated reference genome of Naja naja from Suryamohan et al.~from an
  dindian locality
\item
  A long scaffold query genome from a sri lankan individual of Naja naja
\end{enumerate}

\subsection{Mapping the query genome to the refernece genome to attain a
mapped genome with the same chromosome
names}\label{mapping-the-query-genome-to-the-refernece-genome-to-attain-a-mapped-genome-with-the-same-chromosome-names}

I used ragtag for this (Alonge, Michael, et al.~``Automated assembly
scaffolding elevates a new tomato system for high-throughput genome
editing.'' Genome Biology (2022).
\url{https://doi.org/10.1186/s13059-022-02823-7})

All you need to do is install a ``ragtag'' environment the same way we
would do with conda (other neccessary modules are python and minimap,
but these are on scw), then use the following code:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#!/bin/bash {-}{-}login}
\CommentTok{\#SBATCH {-}{-}job{-}name=ragdag}
\CommentTok{\#SBATCH {-}{-}output=/scratch/b.lxn25yng/slurmscripts/logs/fst\_ang\_\%J.out}
\CommentTok{\#job stderr file}
\CommentTok{\#SBATCH {-}{-}error=/scratch/b.lxn25yng/slurmscripts/logs/fst\_ang\_\%J.err}
\CommentTok{\#SBATCH {-}{-}partition=htc}
\CommentTok{\#SBATCH {-}{-}time=0{-}12:00}
\CommentTok{\#SBATCH {-}{-}nodes=1}
\CommentTok{\#SBATCH {-}{-}ntasks=10}
\CommentTok{\#SBATCH {-}{-}mem=64G}
\CommentTok{\#SBATCH {-}{-}account=scw2119}
\CommentTok{\#SBATCH {-}{-}chdir=/scratch/b.lxn25yng/slurmscripts/}

\ExtensionTok{module}\NormalTok{ purge}

\CommentTok{\# Load modules}
\ExtensionTok{module}\NormalTok{ load python/3.10.4}
\ExtensionTok{module}\NormalTok{ load minimap2/2.24}

\BuiltInTok{source}\NormalTok{ /scratch/b.lxn25yng/ragtag\_env/bin/activate}

\BuiltInTok{cd}\NormalTok{ /scratch/b.lxn25yng/naja/}

\ExtensionTok{ragtag.py}\NormalTok{ scaffold GCA\_009733165.1\_Nana\_v5\_genomic.fna naja\_sl.fasta }\AttributeTok{{-}o}\NormalTok{ ragtag\_out}
\end{Highlighting}
\end{Shaded}

Now we have a query genome mapped to the original fasta reference
genome(to visualize in igv i cut out all the unasigned scaffolds, too
many to visualize in igv): show dotplot of total alignments

\subsection{Toxin Annotation using
ToxCodAn}\label{toxin-annotation-using-toxcodan}

(Nachtigall et al.~(2024) ToxCodAn-Genome: an automated pipeline for
toxin-gene annotation in genome assembly of venomous lineages. Giga
Science. DOI: \url{https://doi.org/10.1093/gigascience/giad116}):

The following script had the following prerequesits to be installed in
this case on my local machine: - python, biopython and pandas -
NCBI-BLAST (v2.9.0 or above) - Exonerate - Miniprot - GffRead - R

to run toxcodan, i downloaded the elapidae toxin database from toxcodan
github containing 1150 toxic CDS from 76 species.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{wget}\NormalTok{ https://raw.githubusercontent.com/pedronachtigall/ToxCodAn{-}Genome/main/Databases/Elapidae\_db.fasta}
\end{Highlighting}
\end{Shaded}

To run toxcodan i installed it as a conda envirmonment onmy laptop,
workflow cpied from toxcodan github:

If the user wants to install ToxCodAn-Genome and all dependencies using
Conda environment, follow the steps below:

Tip: First, ensure that you have added all conda channels properly in
the following order:

conda config --add channels defaults conda config --add channels
bioconda conda config --add channels conda-forge

Create the environment:

conda create -n ToxcodanGenome -c bioconda python biopython pandas blast
exonerate miniprot gffread hisat2 samtools stringtie trinity spades

Git clone the ToxCodAn-Genome repository and add the bin to your PATH:

git clone \url{https://github.com/pedronachtigall/ToxCodAn-Genome.git}
echo ``export PATH=\(PATH:\)PWD/ToxCodAn-Genome/bin/''
\textgreater\textgreater{} \textasciitilde/.bashrc Replace
\textasciitilde/.bashrc to \textasciitilde/.bash\_profile if needed.

It may be needed to apply ``execution permission'' to all bin
executables in ``ToxCodAn-Genome/bin/'':

chmod +x ToxCodAn-Genome/bin/* Then, run ToxCodAn-Genome as described in
the ``Usage'' section.

To activate the environment to run ToxCodAn-Genome just use the command:

conda activate ToxcodanGenome

To deactivate the environment just use the command:

conda deactivate

Once all works:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{conda}\NormalTok{ activate ToxcodanGenome}
\end{Highlighting}
\end{Shaded}

run toxcodan, twice once for each fasta, i used standard filters of 80\%
identity match, 400 min bp and 50000 max bp as recommended by pedro.
changing these filters had minimal impact on results, non perfectly
matched toxins were numerous mainly due to unexpected start or stop
codons, this could very likely be due to the reference genome containing
many unasigned scaffold:

\begin{Shaded}
\begin{Highlighting}[]
 \ExtensionTok{/home/alexn/Annotating/ToxCodAn{-}Genome/bin/toxcodan{-}genome.py} \AttributeTok{{-}g}\NormalTok{ /home/alexn/Annotating/GCA\_009733165.1\_Nana\_v5\_genomic.fasta }\AttributeTok{{-}d}\NormalTok{ /home/alexn/Annotating/Elapidae\_db.fasta }\AttributeTok{{-}c}\NormalTok{ 6}
\end{Highlighting}
\end{Shaded}

this spits out ToxCodAnGenome-output into the working directory.
remember to change this to e.g.~ToxCodAnGenome-outputNajaS1 then:

\begin{Shaded}
\begin{Highlighting}[]
 \ExtensionTok{/home/alexn/Annotating/ToxCodAn{-}Genome/bin/toxcodan{-}genome.py} \AttributeTok{{-}g}\NormalTok{ /home/alexn/Annotating/ragtag\_najaS2{-}najaS1.fasta }\AttributeTok{{-}d}\NormalTok{ /home/alexn/Annotating/Elapidae\_db.fasta }\AttributeTok{{-}c}\NormalTok{ 6}
\end{Highlighting}
\end{Shaded}

You end up with multiple files, toxin annotation and matched getion gtfs
and their according cds, these can be used with IGV or other
applications. I went on to make a tsv of the toxin annotaion file to
then visualize in karyoplot, i manually filtered the tsv for 3Ftx, CRISP
and SVMP toxins, again twice:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{python}\NormalTok{ ToxCodAn{-}Genome/bin/fromCDStoGENE.py   Naja/ToxCodAnGenome\_outputNajaS1/toxin\_annotation.gtf   Naja/ToxCodAnGenome\_outputNajaS1/toxin\_annotation\_GENE\_NajaS1.tsv}

\ExtensionTok{python}\NormalTok{ ToxCodAn{-}Genome/bin/fromCDStoGENE.py   ToxCodAnGenome\_outputNajaS2/toxin\_annotation.gtf   ToxCodAnGenome\_outputNanaS1/toxin\_annotation\_GENE\_NajaS2.tsv}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{Rscript}\NormalTok{ ToxCodAn{-}Genome/b}
\ExtensionTok{in/PlotToxinLoci.R}   \AttributeTok{{-}i}\NormalTok{ ToxCodAnGenome\_outputNajaS1/toxin\_annotation\_GENE\_NajaS1.tsv   }\AttributeTok{{-}o}\NormalTok{ output}


\ExtensionTok{Rscript}\NormalTok{ ToxCodAn{-}Genome/b}
\ExtensionTok{in/PlotToxinLoci.R}   \AttributeTok{{-}i}\NormalTok{ ToxCodAnGenome\_outputNajaS2/toxin\_annotation\_GENE\_NajaS2.tsv   }\AttributeTok{{-}o}\NormalTok{ output}
\end{Highlighting}
\end{Shaded}

\begin{figure}
\centering
\pandocbounded{\includegraphics[keepaspectratio]{najas1.png}}
\caption{Naja Reference Karyoplot}
\end{figure}

\begin{figure}
\centering
\pandocbounded{\includegraphics[keepaspectratio]{najas2.png}}
\caption{Naja Query KaroPlot}
\end{figure}

\end{document}
